bitwarden_license/src/Scim/Commands/Users/DeleteUserCommand.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+using Bit.Scim.Commands.Users.Interfaces;
+using Bit.Scim.Models;
+
+namespace Bit.Scim.Commands.Users;
+
+public class DeleteUserCommand : IDeleteUserCommand
+{
+    private readonly IOrganizationService _organizationService;
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public DeleteUserCommand(
+        IOrganizationService organizationService,
+        IOrganizationUserRepository organizationUserRepository)
+    {
+        _organizationService = organizationService;
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    public async Task DeleteUserAsync(Guid organizationId, Guid id, ScimUserRequestModel model)
+    {
+        var orgUser = await _organizationUserRepository.GetByIdAsync(id);
+        if (orgUser == null || orgUser.OrganizationId != organizationId)
+        {
+            throw new NotFoundException("User not found.");
+        }
```

Comment:
This logic already exists within the `organizationService.DeleteUserAsync` which makes this a bit repetitive. Should we check what error we get back from the OrganizationService and handle it that way instead?

However by removing that logic we've essentially made this command just a direct call to the service. At which point the commands benefit is slightly unclear. (Other than acting as a potential basis for in the future extracting the `DeleteUserAsync` completely into a command.

Reply:
@MGibson1 curious what your thoughts are, in my opinion we can either.

1. Not create a command for this
2. Create a command for deleting a organization user, but put it in `Core`, and deprecate the method on the service. The thinking being that we will eventually want to migrate over to use the command instead of the service.

Reply:
I vote 2, personally. `DeleteUserAsync` can be a command in `Core` which can be called from wherever, SCIM might just have some error handling code around it, but wouldn't itself implement a separate query.

 Although @Hinton my question would then be how to handle the call to `OrganizationService.HasConfirmedOwnersExceptAsync` in `OrganizationService.DeleteUserAsync`. Is that considered a query?

Reply:
2 sounds good to me although it can turn out to be a bigger refactor than anticipated.
@eliykat I second that question on `OrganizationService.HasConfirmedOwnersExceptAsync` and also `OrganizationService.DeleteAndPushUserRegistrationAsync`. Should I create a new query and command and mark the old methods as obsolete?

Reply:
Honestly, just having the command call the service would be enough. Once everything uses the command we can extract it. That saves us having two identical implementations.

The alternative is to refactor the organization service to call the command, and ensure we only have a single implementation. If code is generic enough it's fine to continue having it in the service.

Reply:
I'll keep this simple and make the command just call the service for now, we can refactor this later when we switch everything to use commands.

There is a slight difference between the implementations. When the user is not found on `OrganizationService` it returns a `BadRequestException` whereas SCIM throws a `NotFoundException`. How do I handle this? Change the thrown exception type to `NotFoundException`for both or handle that inside `ExceptionHandlerFilterAttribute` on SCIM?

Reply:
I've up the check back on the command even though it seems redundant just because it's easier to throw a different exception type.

Reply:
That's fine, it's redundant but it was also already here, and it ensures that we comply with the SCIM protocol without causing regressions in other clients.